File: app/src/main/java/com/example/johnmccormick/pickpocketkotlin/widgets/answerboxes/AnswerBoxContract.java
package com.example.johnmccormick.pickpocketkotlin.widgets.answerboxes;

public interface AnswerBoxContract {

    interface View {
        void showCorrectAtPositionOne();
        void showCorrectAtPositionTwo();
        void showCorrectAtPositionThree();
        void showCorrectAsPositionFour();
        void showRightNumberWrongPositionFour();
        void showRightNumberWrongPositionThree();
        void showRightNumberWrongPositionTwo();
        void showRightNumberWrongPositionOne();
        void showAllCorrect();
        void showIncorrectAtPositionOne();
        void showIncorrectAtPositionTwo();
        void showIncorrectAtPositionThree();
        void showIncorrectAtPositionFour();
    }

    interface Presenter {
        void setView(AnswerBoxContract.View view);
        void setNumberCorrect(int numberCorrectRightPosition, int numberCorrectWrongPosition);
        void setAllCorrect();
        void showAllIncorrect();
    }
}

File: app/src/main/java/com/example/johnmccormick/pickpocketkotlin/widgets/answerboxes/AnswerBoxContract.kt
package com.example.johnmccormick.pickpocketkotlin.widgets.answerboxes

interface AnswerBoxContract {
    interface View {
        fun showCorrectAtPositionOne()
        fun showCorrectAtPositionTwo()
        fun showCorrectAtPositionThree()
        fun showCorrectAtPositionFour()
        fun showRightNumberWrongPositionFour()
        fun showRightNumberWrongPositionThree()
        fun showRightNumberWrongPositionTwo()
        fun showRightNumberWrongPositionOne()
        fun showAllCorrect()
        fun showIncorrectAtPositionOne()
        fun showIncorrectAtPositionTwo()
        fun showIncorrectAtPositionThree()
        fun showIncorrectAtPositionFour()
    }

    interface Presenter{
        fun setView(view: AnswerBoxContract.View)
        fun setNumberCorrect(numberCorrectRightPosition: Int, numberCorrectWrongPosition: Int)
        fun setAllCorrect()
        fun showAllIncorrect()
    }
}

